src/app/pages/common/ix-forms/components/ix-input/ix-input.component.ts

```diff
@@ -46,8 +50,20 @@ export class IxInputComponent implements ControlValueAccessor {
     this.onTouch = onTouched;
   }
 
+  shouldShowToggle(): boolean {
+    return this.type === 'password' && this.togglePw;
+  }
+
   shouldShowResetInput(): boolean {
-    return !this.isDisabled && this.hasValue();
+    return !this.isDisabled && this.hasValue() && !this.shouldShowToggle();
+  }
+
+  getType(): string {
+    return this.type === 'password' ? 'text' : this.type;
+  }
+
+  isSecurity(): boolean {
+    return this.type === 'password' && !this.showPassword;
```

Comment:
I think this can be simplified by only using `type` input.
If `type` is passed in as `password` we always show password toggle (i.e. `togglePw` is not necessary).
Then you could have `inputType` in component and toggle it between `text/password` instead of relying on `security-disc`.

Reply:
Afaik, it is not possible to disable autofill for `input='password'` programmatically. That's why I set the type as text and used `security-disc`. Or this will cause NAS-112711 again.

Reply:
How would it work on non-webkit browsers such as Firefox?

Reply:
Updated it